streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -1046,7 +1047,18 @@ public <VR, KO, VO> KTable<K, VR> leftJoin(final KTable<KO, VO> other,
         return doJoinOnForeignKey(other, foreignKeyExtractor, joiner, TableJoined.with(null, null), materialized, true);
     }
 
-    @SuppressWarnings("unchecked")
+    private final Function<Optional<Set<Integer>>, Integer> getPartition = maybeMulticastPartitions -> {
```

Comment:
This question might be silly and just wondering that should we use Optional for this? 
Because I guess if Set object initialise then cannot be null as long as initialise it null.
So for me not wrapping optional object seems lighter and clear.

But this is very minor thing and can be ignored:)

Reply:
hey @YeonCheolGit . Thanks for the comment. Yeah it looks slightly loaded but the problem is we still want to retain the default behaviour ie when the underneath  partitioner returns null, to invoke default partitioner. This class covers the 3 scenarios: https://github.com/apache/kafka/pull/12803/files#diff-36dd8c03fa5252dbd39042bb49a0d6272728fbd46459c6a45f5189fa59749b32R154-R169. Does that answer your question?

Reply:
@vamossagar12 Thanks for letting me know about your intention:) Your answer is very clear to me.